pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter.go

```diff
@@ -45,7 +46,7 @@ func (admitter *PodEvictionAdmitter) Admit(ar *admissionv1.AdmissionReview) *adm
 	if err != nil {
 		return denied(fmt.Sprintf("kubevirt failed getting the vmi: %s", err.Error()))
 	}
-	if !vmi.IsEvictable() {
```

Comment:
Is `vmi.IsEvictable()` needed in its current form? Don't we always need to check the global KubevirtCR eviction strategy?

Maybe instead of creating `VMIMigratableOnEviction()` it's better to create a `vmi.VMIMigratableOnEviction()` method or simple change `vmi.IsEvictable()`?

Reply:
> Is `vmi.IsEvictable()` needed in its current form? Don't we always need to check the global KubevirtCR eviction strategy?

Correct, no code will use it after this PR.
> 
> Maybe instead of creating `VMIMigratableOnEviction()` it's better to create a `vmi.VMIMigratableOnEviction()` method or simple change `vmi.IsEvictable()`?

I cannot change it or remove it, it's public API for the VMI object, that'd break users with kubevirt as a dependency, plus it might still be useful as an API.

Also I don't think it's a good idea to have such an API call defined on the VMI object, for one this is a KubeVirt mechanism (the whole if cluster has this defined honor it even if not specified in the VMI) so it's not strictly a VMI property and secondly for the same reason it would look weird to have a `vmi.IsEvictable(clusterConfig)` API.

Reply:
> Also I don't think it's a good idea to have such an API call defined on the VMI object, for one this is a KubeVirt mechanism (the whole if cluster has this defined honor it even if not specified in the VMI) so it's not strictly a VMI property and secondly for the same reason it would look weird to have a vmi.IsEvictable(clusterConfig) API.

In my eyes `vmi.IsEvictable(clusterConfig)` is not weird, as it makes sense that a VMI needs extra information in order to provide certain answers. But it's a matter of style I guess :)

Reply:
> I cannot change it or remove it, it's public API for the VMI object, that'd break users with kubevirt as a dependency, plus it might still be useful as an API.

Fair point. It still bothers me though for two reasons:
1) After this PR, `vmi.IsEvictable()` will sometimes return a wrong answer. This might confuse whoever will decide to call it in the future and that's a potential for future bugs.
2) A function that no one uses tends to be unmaintained, which will probably make it even more obsolete over time.

We were just talking about deprecation policy in recent community meeting. @davidvossel @EdDev what is your take here?

IMO the least we can do is document that this function is deprecated and point out the new function you've made.

Reply:
> > I cannot change it or remove it, it's public API for the VMI object, that'd break users with kubevirt as a dependency, plus it might still be useful as an API.
> 
> Fair point. It still bothers me though for two reasons:
> 
>     1. After this PR, `vmi.IsEvictable()` will sometimes return a wrong answer. This might confuse whoever will decide to call it in the future and that's a potential for future bugs.

I think they are 2 very different things, vmi.IsEvictable() is just a (kinda) getter, and as such is just returning a property of the VMI and the VMI alone, this patch doesn't change that, it changes how migrations behave in relation to that field, so for me it would actually be wrong for that function to say that a VMI is evictable when the `EvictionStrategy` is not specified because I'm querying a VMI property alone not if the cluster will actually migrate it on eviction, that's a totally different matter, in fact, a VMI could specify that setting but still not be evictable (it's using an incompatible networking setting for instance).

Also if we turn this API into a function that requires a clusterConfig it'd need a cluster to be up while now this could be used offline for a static tool for example.

Maybe it's just my impression but I see the API as a simple getter of a VMI property, not as something that describes how KubeVirt as a whole reacts to that property, also because if this was the intention from the beginning the API has always been broken (refer to my previous example).  

> 
>     2. A function that no one uses tends to be unmaintained, which will probably make it even more obsolete over time.
It's public API, it's not used in our internal code but it could as well be used in thousands of projects, I agree that there's a risk that it will be unmaintained as you have to grep for it specifically to be aware of it so maybe removing it it's a viable option, but I disagree about modifying it. 

> 
> 
> We were just talking about deprecation policy in recent community meeting. @davidvossel @EdDev what is your take here?
> 
> IMO the least we can do is document that this function is deprecated and point out the new function you've made.

I can definitely add a comment if there is consensus even if I don't see the function as 'useless' since it's still just describing a property of the VMI.

Reply:
You have a point, still, for the same reason, I only see (deprecation and then) removal as a viable strategy.

Reply:
I agree.

What I would want to see here:
* Adding documentation to this method to both clarify it's an attribute getter and to point your new function out.
* Add another method that does the same thing but with different name, e.g. `GetEvictionStrategy()`
  * IsEvictable() can delegate the call to the new method (i.e. calling it with the same parameters)
* Declare this function as deprecated
* According to our deprecation policy (that does not exist yet :smiley:) remove this method entirely.

In the scope of this PR I think that only the first bullet is a must-have.

@EdDev FYI

Reply:
Our API compatibility guarantee is related to api field names and types. Essentially, don't break anything that is stored in a json/yaml field that would cause a live operation to break after an update.

Changing a function name isn't going to break a live operation after update. This function isn't a part of the api schema or anything. It's somewhat frustrating to Developers who update the package and have to modify their code, but I wouldn't say that's a rare occurrence.  Programming interfaces change all the time. 

For example, the standard k8s client code changed a few releases to require a `context` for all API requests. That required us to change every single API call to include a context when we updated to consume the new k8s packages... it was a pain, but important.

I wouldn't stress over this function. Delete it, rename it, or whatever feels best. I just wouldn't keep an old function around that returns inaccurate information.

Reply:
BTW - @acardace - I still think you have a valid point that a function returning an attribute might be valuable here. I'm totally fine with simply renaming it to something like `vmi.GetEvictionStrategy()` if you think it's valuable. your call.